landsatxplore/earthexplorer.py

```diff
@@ -144,8 +185,22 @@ def download(self, identifier, output_dir, dataset=None, timeout=300, skip=False
             entity_id = self.api.get_entity_id(identifier, dataset)
         else:
             entity_id = identifier
-        url = EE_DOWNLOAD_URL.format(
-            data_product_id=DATA_PRODUCTS[dataset], entity_id=entity_id
-        )
-        filename = self._download(url, output_dir, timeout=timeout, skip=skip)
+        # Cycle through the available dataset ids until one works
+        dataset_id_list = DATA_PRODUCTS[dataset]
+        id_num = len(dataset_id_list)
+        for id_count, dataset_id in enumerate(dataset_id_list):
+            try:
+                url = EE_DOWNLOAD_URL.format(
+                    data_product_id=dataset_id, entity_id=entity_id
+                )
+                filename = self._download(
+                    url, output_dir, timeout=timeout, skip=skip, overwrite=overwrite
+                )
+            except EarthExplorerError:
+                if id_count+1 < id_num:
+                    print('Download failed with dataset id {:d} of {:d}. Re-trying with the next one.'.format(id_count+1, id_num))
```

Comment:
Only one thing: This warning is somehow confusing (to me):

```suggestion
                if id_count+1 < id_num:
                    print('Download failed with dataset id {:d} of {:d}. Re-trying with the next one.'.format(id_count+1, id_num))
```

In my test (https://github.com/yannforget/landsatxplore/pull/97#issuecomment-1498157850) it shows up while all scenes seem to be found and downloaded.

It seems to appear with every download, not sure why... Maybe an index issue?

Reply:
This is related to the API policy which restricts access to the download-option function to users who have explicitly asked API access. The details are in this commit https://github.com/yannforget/landsatxplore/commit/f31685c6b7ac6ee3aa207c5b486230d27fe6384f. If one could access this function without special permissions, one could always use the download-option function and get the correct dataset-id there. Unfortunately this is not the case, and to avoid forcing all users to ask for API access, the compromise solution was to hard-code all known dataset ids into the library and try different ids until one works. 

I understand that printing those statements may be confusing, and probably it would be better to just try all dataset ids, and just in case each and every one of them fails, raise an error or a warning to communicate to the user that the dataset id list needs to be updated by the maintainer.